Sources/KituraWebSocket/WSSocketProcessor.swift

```diff
@@ -108,6 +108,8 @@ class WSSocketProcessor: IncomingSocketProcessor {
     /// Close the socket and mark this handler as no longer in progress.
     public func close() {
         handler?.prepareToClose()
+        connection.processor = nil
```

Comment:
So I killing the processor here means it can be closed before prepareToClose completes. Instead I suggest:
1. make `active` in `connection` internal,
2. replace these lines with:
```swift
if connection.active {
    connection.connectionClosed(reason: .closedAbnormally)
}
```
above `handler?.prepareToClose()`

Reply:
Actually `active` is a `private` variable in `WebSocketConnection`. So I coded the other way as I didn't want to change access levels for any variable.

Above changes makes sense. I'll push these changes.

Reply:
@Andrew-Lees11  Another issue just came to my mind after pushing the suggested changes. The issue is that connection is holding a strong reference to processor and processor is holding a strong reference to connection. It becomes a cyclic reference and these objects will never be released.

I'm not expert in Swift. Could you please validate my above theory?

Reply:
I missed the part that processor is a `weak`. Thanks @Andrew-Lees11